include/seastar/core/sstring.hh

```diff
@@ -320,7 +320,18 @@ public:
         *this = std::move(ret);
         return *this;
     }
-
+#if __cplusplus > 202002L
+    /**
+     *  Resize string and use the specified @c op to modify the content and the length
+     *  @param n  new size
+     *  @param op the function object used for setting the new content of the string
+     */
+    template<class Operation>
+    void resize_and_overwrite(size_t n, Operation op) {
+        resize(n);
+        resize(std::move(op)(data(), n));
```

Comment:
This should be optimized to use initialized_later to avoid double-initializing.

Perhaps (pseudocode)

```
   if (capacity() < n) {
       operator=(sstring(initialized_later, n));
   }
   _size = n;
   resize(op(begin(), n));
}
```